app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/ProfileController.scala

```diff
@@ -68,12 +70,44 @@ class ProfileController @Inject() (
       }
   }
 
-  def doesProfileExist(eori: String): Action[AnyContent] = identify.async {
+  def doesProfileExist(eori: String): Action[AnyContent] = identify.async { implicit request =>
     profileRepository
       .get(eori)
-      .map {
-        case Some(_) => Ok
-        case None    => NotFound
+      .flatMap {
+        case Some(_) => Future.successful(Ok)
+        case None    =>
+          customsDataStoreConnector.getEoriHistory(eori).flatMap {
+            case Some(eoriHistoryResponse) =>
+              if (eoriHistoryResponse.eoriHistory.isEmpty) Future.successful(NotFound)
+              else {
+
+                val latestEoriResult = profileRepository.get(eoriHistoryResponse.eoriHistory.head.eori).flatMap {
+                  case Some(historicProfile) =>
+                    for {
+                      updateResult <- profileRepository.updateEori(historicProfile.eori, eori)
+                      _            <- recordsRepository.deleteRecordsByEori(historicProfile.eori)
```

Comment:
this may sound like a silly question - but why are we deleting records? don't we just need to change the eori?

Reply:
@AmarkumarJagtap said that we gonna fetch data again.